internal/dag/status_test.go

```diff
@@ -2818,6 +2821,60 @@ func TestGatewayAPIDAGStatus(t *testing.T) {
 		}},
 	})
 
+	run(t, "spec.rules.forwardTo.serviceName invalid on two matches", testcase{
+		objs: []interface{}{
+			gateway,
+			kuardService,
+			&gatewayapi_v1alpha1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "basic",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app": "contour",
+					},
+				},
+				Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha1.Hostname{
+						"test.projectcontour.io",
+					},
+					Rules: []gatewayapi_v1alpha1.HTTPRouteRule{{
+						Matches: []gatewayapi_v1alpha1.HTTPRouteMatch{{
+							Path: gatewayapi_v1alpha1.HTTPPathMatch{
+								Type:  "Prefix",
+								Value: "/",
+							},
+						}},
+						ForwardTo: []gatewayapi_v1alpha1.HTTPRouteForwardTo{{
```

Comment:
In the case that these two rules are in the reverse order, do we expect the condition message to be `Service "invalid-two" does not exist, Service "invalid-one" does not exist`

Would there be value in ensuring the order of the parts of the message are sorted for comparisons between existing and potentially updated status?

Reply:
Yeah I'd expect if you reverse the order the messages would be swapped. 

if there was an existing condition with the same `Type` and something got added, it should get recreated from scratch so I'd expect the order to be consistent, unless I'm misunderstanding.